### ISSUE TITLE
False Positive PT012 for async iterator

### ISSUE THREAD

**Issue by nmzgnv** [astral-sh/ruff#9730]
## What's wrong
I want to test that async iterator raises error.
```python
with pytest.raises(KeyError, match='unknown'):
    async for _ in gpt.generate(gpt_request):
        pass
```
Receive unexpected error `PT012 'pytest.raises()' block should contain a single simple statement`, but the statement is already simple.

## How it should work

not raise PT012 for simple async for loops as with empty context managers https://github.com/m-burst/flake8-pytest-style/blob/master/docs/rules/PT012.md

## System information

* Operating system: mac OS Ventura 13.5
* Python version: 3.11.4
* ruff version: v0.1.9

**Comment by charliermarsh** [comment]
I agree this is reasonable.

**Comment by charliermarsh** [comment]
Is it possible to rewrite this with a single `await asyncio.gather`?

**Comment by nmzgnv** [comment]
I have no idea about `asyncio.gather`, but if error occures on the first iteration, you can use `await anext(...)`:
```python
async def async_gen():
    for i in range(10):
        raise ValueError('Hello')
        yield i


async def main():
    await anext(async_gen())
```
if an error occurs after many iterations, then you need to go through the iterator completely before the error occurs. Another tric: 
```python
async def main():
    _ = [elem async for elem in async_gen()]
```

**Comment by charliermarsh** [comment]
@RonnyPfannschmidt - Do you have an opinion on this one, before I consider changing the rule?

**Comment by RonnyPfannschmidt** [comment]
I'm a bit torn on this one

The most beautiful Syntax is the false positive

The listcomp makes the intent more clear

This is one of the cases where I wish thered be a consume helper

**Comment by jelle-openai** [comment]
I think this is a false positive and should be fixed. The workarounds proposed here make the code harder to understand.
